newIDE/app/src/guidelines/index.js

```diff
@@ -0,0 +1,205 @@
+// @flow
+import React, { PureComponent } from 'react';
+import ReactDOM from 'react-dom';
+import Button from '@material-ui/core/Button';
+import Popper from '@material-ui/core/Popper';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import guidelines from './data.js';
+import Window from '../Utils/Window';
+import { getHelpLink } from '../Utils/HelpLink';
+import { Line } from '../UI/Grid';
+
+//TODO wrap l'image dans le Paper avec l'overflow hidden
+//Actuellement l'image dépasse du Paper alors qu'il y a une marge autour du Popper
+//Actualisé la position de la Popper.
+
+const styles = {
+  container: {
+    maxWidth: 400,
+    padding: 10,
+    marginTop: 10,
+    overflow: 'hidden',
+  },
+  description: {
+    paddingBottom: 10,
+  },
+};
+
+type State = {|
+  open: boolean,
+  index: number,
+  indexData: number,
+  indexMax: number,
+  anchor: *,
+  arrowRef: *,
+|};
+
+type Props = {|
+  open: boolean,
+  closeHandler: () => void,
+|};
+
+//TODO
+//PureComponent ou React.Component<Props, State> voir la doc pour comprendre la différence.
+//Le bouton restart du tuto ferme le Popper mais le reouvre pas et surtout ne reset pas le Popper a l'index 0...
+
+export default class GuidelinePopOver extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      open: props.open,
+      index: 0,
+      indexData: 0,
+      indexMax: guidelines.length - 1,
+      anchor: null,
+      arrowRef: null,
+    };
+  }
+
+  componentWillReceiveProps(newProps: *) {
+    let elementHTML = document.querySelectorAll(
+      '.guideline-' + guidelines[this.state.indexData].positionBind
+    )[0];
+    const element = ReactDOM.findDOMNode(elementHTML);
```

Comment:
Beware, you're confusing React with pure HTML/JS. findDOMNode is used to find the DOM element of a React component.
Here, you don't actually need React at all :)

You have the class of the DOM element to find (guideline-xxx), so you:
1. Get the DOM element with `document.querySelectorAll`.
2. This is a *list* so check if it's not empty.
3. If it's not empty, get the first element. (if it is empty, you just return, potentially you can log a warning).
4. This element is... a DOM element already :) So no need for findDOMNode. Just pass the element to `anchor` using this.setState(). The Guidelines component will be re-rendered with the new "anchor", which will be sent to the Popper.

Reply:
Note that you're doing all of this inside "componentWillReceiveProps", called when props are changed. In practice, you want to run this (you can put this in a method `_updatePopover element`) when... indexData is changing :) Actually indexData is a duplicate of index => remove it and use index directly.